system-int-test/test/cmr/system_int_test/search/sitemaps_test.clj

```diff
@@ -1,45 +1,84 @@
 (ns cmr.system-int-test.search.sitemaps-test
-  (:require [clj-http.client :as client]
-            [clj-xml-validation.core :as xmlv]
-            [clojure.java.io :as io]
-            [clojure.string :as string]
-            [clojure.test :refer :all]
-            [cmr.system-int-test.data2.core :as d]
-            [cmr.system-int-test.utils.index-util :as index]
-            [cmr.system-int-test.utils.ingest-util :as ingest]
-            [cmr.system-int-test.utils.tag-util :as tags]
-            [cmr.mock-echo.client.echo-util :as e]
-            [cmr.system-int-test.system :as s]
-            [cmr.transmit.config :as transmit-config]
-            [cmr.umm-spec.models.umm-common-models :as cm]
-            [cmr.umm-spec.test.expected-conversion :as exp-conv]))
+  (:require
+   [clj-http.client :as client]
+   [clj-xml-validation.core :as xmlv]
+   [clojure.java.io :as io]
+   [clojure.string :as string]
+   [clojure.test :refer :all]
+   [cmr.mock-echo.client.echo-util :as e]
+   [cmr.search.services.content-service :as content-service]
+   [cmr.search.site.static :as static]
+   [cmr.system-int-test.data2.core :as d]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.tag-util :as tags]
+   [cmr.transmit.config :as transmit-config]
+   [cmr.umm-spec.models.umm-common-models :as cm]
+   [cmr.umm-spec.test.expected-conversion :as exp-conv]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Constants and general utility functions for the tests
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(def ^:private base-url
-  "We don't call to `(transmit-config/application-public-root-url)`
-   due to the fact that it requires a context and we're not creating
-   contexts for these integration tests, we're simply using an HTTP
-   client."
-   (format "%s://%s:%s/"
-           (transmit-config/search-protocol)
-           (transmit-config/search-host)
-           (transmit-config/search-port)))
-
-(defn- get-response
-  [url-path]
-  (->> url-path
-       (str base-url)
-       (client/get)))
-
+(def ^:private test-collections (atom {}))
 (def ^:private validate-sitemap-index
   (xmlv/create-validation-fn (io/resource "sitemaps/siteindex.xsd")))
-
 (def ^:private validate-sitemap
   (xmlv/create-validation-fn (io/resource "sitemaps/sitemap.xsd")))
 
+(defn- add-token-header
+  ([token]
+   (add-token-header token {}))
+  ([token options]
+   (assoc-in options [:headers transmit-config/token-header] token)))
+
+(defn- get-response
+  ([url-path]
+   (get-response url-path {}))
+  ([url-path options]
+   (-> (transmit-config/application-public-root-url :search)
+       (str url-path)
+       (client/get options))))
+
+(defn- do-permission-assertions
```

Comment:
This is a big chunk of code to copy/paste. I would move it into a single test utility namespace to be used by both of the tests.

Reply:
Moved.